src/cargo/ops/cargo_compile/mod.rs

```diff
@@ -657,8 +662,13 @@ fn traverse_and_share(
         _ => unit.kind,
     };
 
+    let is_target_windows_msvc = target_data
+        .short_name(&unit.kind)
+        .ends_with("-pc-windows-msvc");
```

Comment:
Backtraces still live on windows-gnu, and (probably?) also work on -gnullvm, but are almost certainly **also** extremely dead on windows7-msvc. While both are tier 3 targets we don't promise any level of support for, I believe the most correct implementation would currently be:
```suggestion
    let is_target_windows_msvc = {
        let name = target_data.short_name(&unit.kind);
        // to be correct re: windows7, gnullvm, and whatever other silliness happens
        name.contains("windows") && name.contains("msvc")
    };
```

Reply:
Instead of scanning target names, I think ideally cargo should be looking at `cfg` values to avoid ambiguity. Perhaps something roughly like:

```rust
let cfg = target_data.cfg(unit.kind);
cfg.contains(Cfg::Name("windows")) && cfg.contains(Cfg::KeyPair("target_env", "msvc"))
```

Reply:
oh neat, I didn't realize that exists.

Reply:
Cool! I modified it to use `Cfg`.